### PR TITLE
Activities enum

### PR DESCRIPTION
**Pull request by heimrih**
Activities likes report mark

### CODE REVIEW

**Comment by heimrih** [comment]
ready

File: app/controllers/activities_controller.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
+class ActivitiesController < ApplicationController
+  before_action :logged_in_user, only: %i(create destroy)
+  before_action :correct_user, only: %i(edit update)
+  before_action :find_post, only: %i(create destroy update)
+
+  def index
+    @activity = Activity.all
```
Comment (saobv-2223) [inline app/controllers/activities_controller.rb]: You should rename @activity to @activities

File: app/controllers/activities_controller.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  end
+
+  def new
+    @activity = Activity.new
+  end
+
+  def create
+    @activity = @post.activities.create activity_params.merge(user_id: current_user.id)
```
Comment (saobv-2223) [inline app/controllers/activities_controller.rb]: remove ()

File: app/controllers/activities_controller.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    if @activity.save
+      flash[:success] = t "activity.create"
+      redirect_to post_path(id: params[:post_id])
```
Comment (saobv-2223) [inline app/controllers/activities_controller.rb]: remove ()

File: app/controllers/activities_controller.rb [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+  private
+
+  def activity_params
+    params.require(:activity).permit :field
+  end
+
+  def find_post
+    @post = Post.find_by id: params[:post_id]
+    return if @post.present?
```
Comment (saobv-2223) [inline app/controllers/activities_controller.rb]:
```suggestion
    return if @post
```

File: app/controllers/posts_controller.rb [hunk 5]
```diff
@@ -9,6 +9,9 @@ def index
 
   def show
     Post.update_counters [@post.id], views: Settings.posts.view.inc
+
+    # get all activities and user has activity of this post.
+    @activities = @post.activities.includes(:user)
```
Comment (saobv-2223) [inline app/controllers/posts_controller.rb]: remove ()

File: app/models/activity.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+class Activity < ApplicationRecord
+  enum field: {mark: 0, like: 1, report: 2}
+
+  belongs_to :user
+  belongs_to :post
+
+  scope :field, ->(field) {
+    where(field: fields[field])
+  }
```
Comment (saobv-2223) [inline app/models/activity.rb]:
```suggestion
  scope :field, (lambda |field| do 
    where(field: fields[field])
  end)
```

File: app/views/activities/_activity.html.erb [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+          <%= activity.user.name %>
+        </span>
+        <br/>
+        <span>
+          <%= t "activity.act" %>
+          <%= activity.field %>
+        </span>
+        <br/>
+        <% if logged_in? && @current_user.activities.find_by(id: activity.id).present? %>
+          <%= link_to "Un-#{activity.field}", post_activity_path(post_id: @post.id, id: activity.id), method: "DELETE" %>
```
Comment (saobv-2223) [inline app/views/activities/_activity.html.erb]:
```suggestion
          <%= link_to "Un-#{activity.field}", post_activity_path(post_id: @post.id, id: activity.id), method: :delete %>
```

**Comment by heimrih** [comment]
ready

File: app/controllers/activities_controller.rb [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+  before_action :logged_in_user, only: %i(create destroy)
+  before_action :correct_user, only: %i(edit update)
+  before_action :find_post, only: %i(create destroy update)
+
+  def index
+    @activities = Activity.all
+  end
+
+  def new
+    @activities = Activity.new
```
Comment (saobv-2223) [inline app/controllers/activities_controller.rb]: with this case, Activity.new will create AN objeck. therefor you should use @activity
Using plurality when return > 1 record

File: app/views/activities/_activity.html.erb [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+          <%= activity.field %>
+        </span>
+        <br/>
+        <% if logged_in? && @current_user.activities.find_by(id: activity.id).present? %>
+          <%= link_to "Un-#{activity.field}", post_activity_path(post_id: @post.id, id: activity.id), method: :delete %>
+        <% end %>
+      </div>
+    <% end %>
+    <% else %>
+      <%= t "activity.nil" %>
```
Comment (saobv-2223) [inline app/views/activities/_activity.html.erb]:
```suggestion
  <% else %>
    <%= t "activity.nil" %>
```

**Comment by heimrih** [comment]
ready

**Pull request merged**